### PR TITLE
fix(core): Move error obfuscation to FE (no-changelog)

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

Move error obfuscation to FE based on `obfuscate` property to avoid issues like Sentry receives Internal error

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/PAY-1760/confusing-error-message-and-internal-when-trying-to-call-a-sub#comment-37cab576
https://linear.app/n8n/issue/NODE-1537/sentry-should-received-original-error-message-not-obfuscated-one

### CODE REVIEW

**Comment by ivov** [comment]
LGTM except that I tested the subworkflow case from the [original issue](https://linear.app/n8n/issue/PAY-1760) and I still see it [obfuscated when it should not](https://share.cleanshot.com/5Nbm4Wck). Is that meant to be fixed by this PR or is that a follow-up?

**Comment by michael-radency** [comment]
> LGTM except that I tested the subworkflow case from the [original issue](https://linear.app/n8n/issue/PAY-1760) and I still see it [obfuscated when it should not](https://share.cleanshot.com/5Nbm4Wck). Is that meant to be fixed by this PR or is that a follow-up?

this PR meant to fix issue with errors obfuscated in Sentry, I would suggest to descope this case

**Review by ivov — APPROVED** [review]
Not very familiar with the underlying PR but this looks okay.

**Review by ivov — DISMISSED** [review]
e2e error looks relevant

**Comment by ivov** [comment]
I thought this was covered by Adi's fix? Maybe he has better context to review.

File: packages/core/src/WorkflowExecute.ts
Comment (netroy) [inline packages/core/src/WorkflowExecute.ts:1320]: since many nodes wrap unhandled errors in a `NodeOperationError` (like the notion node), we will not not be obfuscating that error.
Is that what we want to do here?

**Review by netroy — COMMENTED** [review]
This PR doesn't change anything around error reporting, and since the caught error isn't rethrown, it won't be picked up by the error reporter either.

**Comment by netroy** [comment]
If the purpose of this PR is to fix the obfuscation code, then can we please update the PR title accordingly?
I can either rebase https://github.com/n8n-io/n8n/pull/10249 after this is merged, or if you want we can cherry-pick that in here, and close the other PR.

File: packages/core/src/WorkflowExecute.ts
Comment (michael-radency) [reply packages/core/src/WorkflowExecute.ts:1320]: @netroy 
we do have logic in `NodeOperationError` to determine if error should be obfuscated, if so - property `obfuscate` would be set to `true` and on FE based on that we would display `Internal error` instead actual error message

goal of this PR to obfuscate error message based on `obfuscate` property just on FE

**Comment by michael-radency** [comment]
> This PR doesn't change anything around error reporting, and since the caught error isn't rethrown, it won't be picked up by the error reporter either.

we do have such [case](https://n8nio.sentry.io/issues/5446470541/?project=4503924908883968&query=&referrer=issue-stream&sort=freq&statsPeriod=7d&stream_index=3) in Sentry, changes in this PR should resolve it

**Comment by michael-radency** [comment]
> I can either rebase #10249 after this is merged, or if you want we can cherry-pick that in here, and close the other PR.

I do think those changes not needed anymore, we need to add this to `WebhookHelpers` where errors indeed lost

File: packages/core/src/WorkflowExecute.ts
Comment (netroy) [reply packages/core/src/WorkflowExecute.ts:1320]: I'm still unsure how the error from line 1137 will ever make it Sentry considering that we catch it, but neither report it, nor rethrow it.

**Review by netroy — DISMISSED** [review]
Thanks for updating the PR title. It makes more sense now.

I'd still like to rebase #10249, so that we can start tracking unhandled errors from individual nodes.
If that becomes too noise, we can remove it later.

**Comment by michael-radency** [comment]
> I'm still unsure how the error from line 1137 will ever make it Sentry considering that we catch it, but neither report it, nor rethrow it.

@netroy 
I think you are right, but in this case we do not need conditions here and always report error WDYT? I did commit code from your PR to do that

@ivov can you please also review this PR or dismiss your previous review?

**Comment by netroy** [comment]
> I did commit code from your PR to do that

I updated the PR. can you copy over the code again please 🙏🏽

**Comment by michael-radency** [comment]
> I updated the PR. can you copy over the code again please 🙏🏽

@netroy done

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**
